packages/server/lib/controllers/mcp/server.ts

```diff
@@ -0,0 +1,144 @@
+import { Server } from '@modelcontextprotocol/sdk/server/index.js';
+import { CallToolRequestSchema, ListToolsRequestSchema } from '@modelcontextprotocol/sdk/types';
+import tracer from 'dd-trace';
+
+import { OtlpSpan, defaultOperationExpiration, logContextGetter } from '@nangohq/logs';
+import { configService, getActionsByProviderConfigKey } from '@nangohq/shared';
+import { truncateJson } from '@nangohq/utils';
+
+import { getOrchestrator } from '../../utils/utils.js';
+
+import type { CallToolRequest, CallToolResult, Tool } from '@modelcontextprotocol/sdk/types';
+import type { Config } from '@nangohq/shared';
+import type { DBConnectionDecrypted, DBEnvironment, DBSyncConfig, DBTeam } from '@nangohq/types';
+import type { Span } from 'dd-trace';
+import type { JSONSchema7 } from 'json-schema';
+
+export async function createMcpServerForConnection(account: DBTeam, environment: DBEnvironment, connection: DBConnectionDecrypted, providerConfigKey: string) {
```

Comment:
all new functions should return a `Result`. It makes error handling more explicit